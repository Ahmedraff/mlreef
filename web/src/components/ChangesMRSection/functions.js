import { Base64 } from 'js-base64';
import Base64ToArrayBuffer from 'base64-arraybuffer';
import { getFileDifferences } from '../../functions/apiCalls';
import CommitsApi from 'apis/CommitsApi';
import { isImageFormat } from 'functions/dataParserHelpers';

const commitsApi = new CommitsApi();

export const fetchFileDiffInfo = (projectId, aheadCommits) => Promise.all(aheadCommits.map(async (commit, index) => {
  const { id: commitId } = commit;
  const commitDiffs = await commitsApi.getCommitDiff(projectId, commitId, 0, false);
  return await Promise.all(commitDiffs?.body?.map(async (commitDiff) => {
    const {
      previousVersionFile,
      nextVersionFile,
      imageFileSize,
    } = await getFileDifferences(projectId, commitDiff, commit.parent_ids[0], commit.id);
    let previousVersionFileParsed = previousVersionFile;
    let nextVersionFileParsed = nextVersionFile;
    let sizeDeleted = 0;
    let sizeAdded = 0;
    const fileName = commitDiff.old_path.split('/').slice(-1)[0];
    if (previousVersionFile && !isImageFormat(fileName)) {
      previousVersionFileParsed = Base64.decode(
        Base64ToArrayBuffer.encode(previousVersionFile),
      );
      sizeDeleted = previousVersionFile.byteLength;
    }
    if (nextVersionFile && !isImageFormat(fileName)) {
      nextVersionFileParsed = Base64.decode(Base64ToArrayBuffer.encode(nextVersionFile));
      sizeAdded = nextVersionFile.byteLength;
    }
    const path = commitDiff.new_path
      ? commitDiff.new_path
      : commitDiff.old_path;
    const fileExtension = path.split('.').pop().toLowerCase();
    return {
      fileName,
      fileExtension,
      previousVersionFileParsed,
      nextVersionFileParsed,
      id: `${commit.id} ${index}`,
      sizeDeleted,
      sizeAdded,
      imageFileSize,
    };
  }));
}));
