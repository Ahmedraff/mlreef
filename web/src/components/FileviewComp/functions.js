import FilesApi from 'apis/FilesApi.ts';
import CommitsApi from 'apis/CommitsApi.ts';

const commitsApi = new CommitsApi();

const filesApi = new FilesApi();

const getCommit = (projectId, lastCommitId) => commitsApi
  .getCommitDetails(projectId, lastCommitId);

const getFileAndInformation = (gid, file, branch, commit) => filesApi
  .getFileData(gid, file?.includes('/') ? encodeURIComponent(file) : file, branch || commit)
  .then(async (fData) => {
    const commitInfoDet = await getCommit(gid, fData.last_commit_id);
    return {
      fData,
      commitInfoDet,
    };
  });

export default {
  getFileAndInformation,
};
