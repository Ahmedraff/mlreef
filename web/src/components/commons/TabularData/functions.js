import genUUID from 'uuid/v1';

const parseNumber = (text) => {
  const num = parseFloat(text);

  return Number.isNaN(num) ? text : num;
};

export const transpose = (matrix) => matrix[0].map((_, ci) => matrix.map((row) => row[ci]));

const sanatizeMatrix = (matrix) => {
  const head = matrix[0];
  const tail = matrix[matrix.length - 1];

  const sameLength = head.length === tail.length;
  const lastRowfilled = tail.every((col) => col !== undefined);

  return sameLength && lastRowfilled ? matrix : matrix.slice(0, -1);
};

export const cvsToArray = (text, separator = ',') => {
  const matrix = text.split('\n')
    .map((line) => line.split(separator)
      .map(parseNumber));

  return sanatizeMatrix(matrix);
};

export const jsonToArray = (input) => {
  const items = Array.isArray(input) ? input : Object.values(input);

  if (!items.length) return items;
  // const row = Object.values(input)[0];
  const row = items[0];
  const header = Object.keys(row).map((label) => label);

  // const data = Object.values(input)
  const data = items
    .map((cols) => Object.values(cols).map((val) => val));

  const matrix = [header].concat(data);

  return sanatizeMatrix(matrix);
};

export const arrayToRichData = (matrix, separator = ',') => {
  const header = matrix[0];

  return {
    id: genUUID(),
    meta: {
      separator,
      timestamp: new Date(),
    },
    matrix,
    // data: flat(matrix.map((row, y) => row.map((value, x) => ({ x, y, value }))))
    data: matrix.map((cols, y) => ({
      id: y,
      cols: cols.map((value, x) => ({
        x,
        y,
        value,
        label: header[x],
      })),
    })),
  };
};
