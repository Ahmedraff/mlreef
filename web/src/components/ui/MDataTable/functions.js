const extractField = (row, x) => row.cols.find((col) => col.x === x)?.value;

export const sortDesc = (x) => (rows) => rows.slice()
  .sort((a, b) => extractField(b, x) - extractField(a, x));

export const sortAsc = (x) => (rows) => rows.slice()
  .sort((a, b) => extractField(a, x) - extractField(b, x));

export const round = (decs) => (input) => {
  const decimals = parseInt(decs, 10);

  if (Number.isNaN(decimals)) return input;

  if (decimals > 12) return input;

  const num = parseFloat(input);
  if (Number.isNaN(num)) return input;

  return Math.round(num * (10 ** decimals)) / (10 ** decimals);
};
