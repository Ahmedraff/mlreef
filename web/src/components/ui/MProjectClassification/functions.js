import { ALGORITHM, DATA_OPERATION, MODEL, OPERATION, VISUALIZATION } from "dataTypes";

export const getSearchable = (feSearchableTypes) => {
  switch (feSearchableTypes) {
    case DATA_OPERATION:
      return OPERATION;
    case MODEL:
      return ALGORITHM;
    default:
      return VISUALIZATION;
  }
}