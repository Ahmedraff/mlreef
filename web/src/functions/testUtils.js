import { createStore, applyMiddleware } from 'redux';

import rootReducer from '../reducers/index';
import { middlewares } from '../configureStore';

// eslint-disable-next-line import/prefer-default-export
export const storeFactory = (initialState) => {
  const createStoreWithMiddleware = applyMiddleware(...middlewares)(createStore);
  return createStoreWithMiddleware(rootReducer, initialState);
};
