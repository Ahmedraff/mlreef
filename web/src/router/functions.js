/**
 * Compose like: f.g x = g(f(x))
 *
 * @example compose (fn4, fn3, fn2, fn1) (arg) === fn4(fn3(fn2(fn1(arg))))
 *
 * @param {Array[Function]} fns sequence of functions.
 * @param arg input parameter.
 *
 * @return far left funcion return.
 */
export const c = (...fns) => (arg) => fns.reverse().reduce((acc, fn) => fn(acc), arg);

/**
 * Return the recorded route that matches name.
 */
export const findRouteInfo = (routes) => ({ name }) => routes.find((route) => route.name === name);

/**
 * Replace placeholders by parameters. e.g. /:user/messages -> /nick/messages.
 *
 * @param {Object(Route)} to implicit.
 * @param {Object} params e.g. key-val object, from the to object.
 * @param {Object(Route)} record the route from routes.
 *
 * @return {Object(Route)} route with path modified.
 */
export const replaceParams = (to) => (record = {}) => {
  if (to.params) {
    const path = Object.entries(to.params)
      .reduce((final, [k, v]) => final.replace(new RegExp(`:${k}`), v), record.path || '');

    return { ...record, path };
  }

  return record;
};

/**
 * Add the hash part. e.g. #last-menu-item
 *
 * @param {Object(Route)} to implicit.
 * @param {String} hash e.g. last-menu-item, from the to object.
 * @param {Object(Route)} record the route from routes.
 *
 * @return {Object(Route)} route with path modified.
 */
export const addHash = (to) => (record) => !to.hash ? record
  : { ...record, path: `${record.path}#${to.hash}` };

export const makeQuery = (query) => Object.entries(query)
  .map(([k, v]) => `${k}=${v}`)
  .join('&');

/**
 * Add the query search. e.g. ?user=nickname&priority=high
 *
 * @param {Object(Route)} to implicit.
 * @param {Object} query e.g. key-val object, from the to object.
 * @param {Object(Route)} record the route from routes.
 *
 * @return {Object(Route)} route with path modified.
 */
export const addQuery = (to) => (record) => !to.query ? record
  : { ...record, path: `${record.path}?${makeQuery(to.query)}` };

export const unpack = ({ path }) => path;

export const resolveRoute = (routes, to) => c(
  unpack, addQuery(to), addHash(to), replaceParams(to), findRouteInfo(routes),
)(to);
