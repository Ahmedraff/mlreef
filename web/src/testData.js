export const imagesToRender = [
  {
    commitId: 'd80ab8ff15fb1f327bac0f5679b454cbec247759',
    parentCommitId: '6819ee9d2e8f7796887cc87af2d396c9894a2ebf',
    previousImage: {},
    nextImage: {},
    fileName: 'color_0_0054_A.png',
  },
];

export const mockMatchDataCommitDet = {
  path: '/my-projects/:projectId/commit/:commitId',
  url: '/my-projects/12395599/commit/aede036f526ea5d591f5ee56c131ed8472483912',
  isExact: true,
  params: {
    projectId: '12395599',
    branch: 'master',
    file: '.gitignore',
    commitId: 'aede036f526ea5d591f5ee56c131ed8472483912',
  },
};

export const commitMockObject = {
  id: 'd80ab8ff15fb1f327bac0f5679b454cbec247759',
  short_id: 'd80ab8ff',
  created_at: '2019-11-09T14:24:40.000Z',
  parent_ids: [
    '6819ee9d2e8f7796887cc87af2d396c9894a2ebf',
  ],
  title: 'Replace color_0_0054_A.png',
  message: 'Replace color_0_0054_A.png',
  author_name: 'cpmlreef',
  author_email: 'cp@mlreef.com',
  authored_date: '2019-11-09T14:24:40.000Z',
  committer_name: 'cpmlreef',
  committer_email: 'cp@mlreef.com',
  committed_date: '2019-11-09T14:24:40.000Z',
  stats: {
    additions: 0,
    deletions: 0,
    total: 0,
  },
  status: null,
  last_pipeline: null,
  project_id: 14448940,
};

export const mockDiffArray = [
  {
    old_path: 'showcase/color_0_0054_A.png',
    new_path: 'showcase/color_0_0054_A.png',
    a_mode: '100644',
    b_mode: '100644',
    new_file: false,
    renamed_file: false,
    deleted_file: false,
    diff: 'Binary files a/showcase/color_0_0054_A.png and b/showcase/color_0_0054_A.png differ\n',
  },
];

export const projectsArrayMock = {
  projects: {
    all: [
      {
        id: 14448940,
        description: 'Categorical Classification of alphabets in the American Sign Language',
        name: 'Sign Language Classifier',
        name_with_namespace: 'MLReef / Sign Language Classifier',
        path: 'sign-language-classifier',
        path_with_namespace: 'mlreef/sign-language-classifier',
        created_at: '2019-09-23T08:51:23.983Z',
        default_branch: 'master',
        tag_list: [],
        ssh_url_to_repo: 'git@gitlab.com:mlreef/sign-language-classifier.git',
        http_url_to_repo: 'https://gitlab.com/mlreef/sign-language-classifier.git',
        web_url: 'https://gitlab.com/mlreef/sign-language-classifier',
        readme_url: 'https://gitlab.com/mlreef/sign-language-classifier/blob/master/README.md',
        avatar_url: 'https://gitlab.com/uploads/-/system/project/avatar/14448940/proxy.duckduckgo.png',
        star_count: 0,
        forks_count: 0,
        last_activity_at: '2019-11-10T15:14:17.804Z',
        namespace: {
          id: 5351747,
          name: 'MLReef',
          path: 'mlreef',
          kind: 'group',
          full_path: 'mlreef',
          parent_id: null,
          avatar_url: '/uploads/-/system/group/avatar/5351747/MLReef_Logo_POS_H_icon-01.png',
          web_url: 'https://gitlab.com/groups/mlreef',
        },
      },
    ],
    selectedProject: {
      id: 12395599,
      description: 'This is MLReefÂ´s demo repository.',
      name: 'demo',
      name_with_namespace: 'MLReef / demo',
      path: 'mlreef-demo',
      path_with_namespace: 'mlreef/mlreef-demo',
      created_at: '2019-05-18T09:34:29.346Z',
      default_branch: 'master',
      tag_list: [],
      ssh_url_to_repo: 'git@gitlab.com:mlreef/mlreef-demo.git',
      http_url_to_repo: 'https://gitlab.com/mlreef/mlreef-demo.git',
      web_url: 'https://gitlab.com/mlreef/mlreef-demo',
      readme_url: 'https://gitlab.com/mlreef/mlreef-demo/blob/master/README.md',
      avatar_url: 'https://gitlab.com/uploads/-/system/project/avatar/12395599/shutterstock_129655604-Converted-copy-text.png',
      star_count: 2,
      forks_count: 1,
      last_activity_at: '2019-11-11T10:36:06.189Z',
      namespace: {
        id: 5351747,
        name: 'MLReef',
        path: 'mlreef',
        kind: 'group',
        full_path: 'mlreef',
        parent_id: null,
        avatar_url: '/uploads/-/system/group/avatar/5351747/MLReef_Logo_POS_H_icon-01.png',
        web_url: 'https://gitlab.com/groups/mlreef',
      },
    },
  },
  users: [
    {
      id: 3839940,
      name: 'cpmlreef',
      username: 'cpmlreef',
      state: 'active',
      avatar_url: 'https://assets.gitlab-static.net/uploads/-/system/user/avatar/3839940/avatar.png',
      web_url: 'https://gitlab.com/cpmlreef',
    },
  ],
};


export const branchesMock = [
  {
    name: 'data-instance/664e1c60',
    commit: {
      id: '12e05078d142b2bee947e86d6a52db4a5f85eb65',
      short_id: '12e05078',
      created_at: '2019-11-01T21:08:23.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-01T21:08:23.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-01T21:08:23.000+00:00',
    },
    merged: false,
    protected: false,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: false,
  },
  {
    name: 'data-instance/9ae9c8c0',
    commit: {
      id: '3e36bfab31a1d43457e00f46af9e6d31d4f55a66',
      short_id: '3e36bfab',
      created_at: '2019-11-14T20:32:52.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-14T20:32:52.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-14T20:32:52.000+00:00',
    },
    merged: false,
    protected: false,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: false,
  },
  {
    name: 'master',
    commit: {
      id: '3e36bfab31a1d43457e00f46af9e6d31d4f55a66',
      short_id: '3e36bfab',
      created_at: '2019-11-14T20:32:52.000+00:00',
      parent_ids: null,
      title: 'Add pipeline results [skip ci]',
      message: 'Add pipeline results [skip ci]',
      author_name: 'mlreefdemo',
      author_email: 'rainer+mlreefdemo@systemkern.com',
      authored_date: '2019-11-14T20:32:52.000+00:00',
      committer_name: 'mlreefdemo',
      committer_email: 'rainer+mlreefdemo@systemkern.com',
      committed_date: '2019-11-14T20:32:52.000+00:00',
    },
    merged: false,
    protected: true,
    developers_can_push: false,
    developers_can_merge: false,
    can_push: true,
    default: true,
  },
];
