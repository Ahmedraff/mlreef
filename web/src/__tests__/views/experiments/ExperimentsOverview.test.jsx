import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { projectsArrayMock, experimentsClassifiedMock } from 'testData';
import { storeFactory } from 'functions/testUtils';
import ExperimentsOverview from 'components/experiments-overview/ExperimentOverview';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

const mockPush = jest.fn();

const setup = () => {
  const store = storeFactory({ projects: projectsArrayMock.projects });
  const wrapper = shallow(
    <ExperimentsOverview
      store={store}
      history={{ push: mockPush }}
    />,
  );
  const afterDive = wrapper.dive().dive();

  return afterDive;
};

/* describe('test presence of elements in DOM', () => {
  test('compare with the latest snapshot of the empty view', () => {
    const store = storeFactory({ projects: projectsArrayMock.projects });
    const component = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter key="rerere">
            <ExperimentsOverview history={{ push: mockPush }} />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
}); */

describe('test functionality', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
    wrapper.setState({ experiments: experimentsClassifiedMock, all: experimentsClassifiedMock });
  });

  test('assert that top section button callbacks', () => {
    const handleButtonsClickMock = jest.fn();
    wrapper.instance().handleButtonsClick = handleButtonsClickMock;
    const ids = [
      'all',
      'running',
      'success',
      'failed',
      'canceled',
    ];
    ids.forEach((id) => {
      const mockedEv = { target: { id } };
      wrapper.find(`button#${id}`).simulate('click', mockedEv);
      expect(handleButtonsClickMock.mock.calls.length).toBe(1);
      expect(handleButtonsClickMock).toHaveBeenCalledWith(mockedEv);
      handleButtonsClickMock.mockClear();
    });
  });

  test('assert that new experiment is called with the component function', () => {
    const { projects: { selectedProject: { namespace, slug } } } = projectsArrayMock;
    wrapper.find('#new-experiment').dive().find('button').simulate('click');
    expect(mockPush.mock.calls.length).toBe(1);
    expect(mockPush).toHaveBeenCalledWith(`/${namespace}/${slug}/-/experiments/new`);
  });

  test('assert that filters work', () => {
    const handleButtonsClickMock = jest.fn();
    wrapper.instance().handleButtonsClick = handleButtonsClickMock;
    const id = 'failed';
    const mockedEv = { target: { id } };
    wrapper.find(`button#${id}`).simulate('click', mockedEv);
    expect(wrapper.find('ExperimentCard')).toHaveLength(1);
  });
});
