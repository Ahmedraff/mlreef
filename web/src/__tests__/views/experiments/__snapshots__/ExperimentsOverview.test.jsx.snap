// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test presence of elements in DOM compare with the latest snapshot of the empty view 1`] = `
<div
  id="experiments-overview-container"
>
  <div
    className="navbar"
  >
    <div
      className="w-100 px-3 d-flex"
    >
      <div
        className="my-auto"
      >
        <a
          href="/explore"
          onClick={[Function]}
        >
          <img
            alt=""
            className="logo"
            src={
              Object {
                "canInstrument": true,
                "createTransformer": [Function],
                "getCacheKey": [Function],
                "process": [Function],
              }
            }
          />
        </a>
      </div>
      <div
        className="m-dropdown  ml-3 my-auto"
      >
        <div
          className="m-dropdown-button"
        >
          <button
            className="btn btn-dark px-1 "
            onClick={[Function]}
            type="button"
          >
            Projects
            <i
              className="fa fa-chevron-down ml-2 my-auto"
            />
          </button>
        </div>
        <div
          className="m-dropdown-list-container border-rounded  mt-1"
          onClick={[Function]}
        >
          <div
            className="project-box"
          >
            <div
              className="user-projects"
            >
              <p>
                <a
                  href="/explore"
                  onClick={[Function]}
                >
                  Explore projects
                </a>
              </p>
            </div>
            <div
              className="project-search"
            >
              <input
                placeholder="Search your projects"
                type="text"
              />
              <div
                className="mt-3"
              >
                <b>
                  Frequently visited
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="m-dropdown  ml-3 my-auto"
      >
        <div
          className="m-dropdown-button"
        >
          <button
            className="btn btn-dark px-1 "
            onClick={[Function]}
            type="button"
          >
            Groups
            <i
              className="fa fa-chevron-down ml-2 my-auto"
            />
          </button>
        </div>
        <div
          className="m-dropdown-list-container border-rounded  mt-1"
          onClick={[Function]}
        >
          <div
            className="project-box"
          >
            <div
              className="user-projects"
            >
              <p>
                Explore Groups
              </p>
            </div>
            <div
              className="project-search"
            >
              <input
                placeholder="Search your groups"
                type="text"
              />
              <div
                className="mt-3"
              >
                <b>
                  Frequently visited
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="d-none d-lg-flex ml-auto my-auto"
      >
        <a
          className="btn btn-sm btn-dark mr-3 px-3"
          href="/login?redirect=goback"
          onClick={[Function]}
        >
          Sign in
        </a>
        <a
          className="btn btn-sm btn-outline-secondary keep-border"
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </div>
    </div>
  </div>
  <div
    className="project-container"
    style={
      Object {
        "backgroundColor": "#e5e5e5",
      }
    }
  >
    <div
      className="project-details main-content"
    >
      <div
        className="project-nav"
      >
        <p
          className="px-1"
        >
           my-namespace &gt; 
        </p>
        <a
          href="/my-namespace/the-project-name"
          onClick={[Function]}
        >
          <p
            className="px-1"
          >
             demo &gt; 
          </p>
        </a>
        <p
          className="px-1"
        >
           Experiments 
        </p>
      </div>
      <div
        className="project-info"
      >
        <div
          className="project-id"
        >
          <a
            href="/my-namespace/the-project-name"
            onClick={[Function]}
          >
            <div
              className="project-pic overflow-hidden"
            >
              <img
                alt=""
                src="https://gitlab.com/uploads/-/system/project/avatar/12395599/shutterstock_129655604-Converted-copy-text.png"
                style={
                  Object {
                    "minWidth": "100%",
                  }
                }
              />
            </div>
          </a>
          <div
            className="project-name"
          >
            <a
              href="/my-namespace/the-project-name"
              id="projectName"
              onClick={[Function]}
            >
              demo
            </a>
            <p
              id="projectId"
            >
              Project ID:
              this-is-the-mlreef-uuid
               
              | 
              0
               used
            </p>
          </div>
        </div>
        <div
          className="project-options"
        >
          <div
            className="options d-flex mr-2"
          >
            <button
              className="option-name btn btn-hidden border-rounded-left py-2 px-3 my-0"
              type="button"
            >
              <img
                alt=""
                className="mr-0 mr-lg-1"
                id="option-image"
                src="/images/svg/star_01.svg"
              />
              <span
                className="my-auto d-none d-lg-block"
              >
                Star
              </span>
            </button>
            <div
              className="counter border-rounded-right h-100"
            >
              <span>
                2
              </span>
            </div>
          </div>
          <div
            className="options d-flex mr-2"
          >
            <button
              className="option-name btn btn-hidden border-rounded-left py-2 px-3 my-0"
              onClick={[Function]}
              type="button"
            >
              <img
                alt=""
                className="mr-0 mr-lg-1"
                id="option-image"
                src="/images/svg/fork_01.svg"
              />
              <span
                className="my-auto d-none d-lg-block"
              >
                Fork
              </span>
            </button>
            <div
              className="counter border-rounded-right h-100"
            >
              <span>
                1
              </span>
            </div>
          </div>
          <div
            className="options d-flex"
          >
            <div
              className="option-name border-rounded-left py-2 px-3 my-0"
            >
              <img
                alt=""
                className="mr-0 mr-lg-1"
                id="option-image"
                src="/images/svg/clone_01.svg"
              />
              <span
                className="my-auto d-none d-lg-block"
              >
                Clone
              </span>
            </div>
            <div
              className="border-rounded-right h-100 counter clone-dropdown"
              id="t-clonedropdown-toggle"
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              <span
                className="fa fa-chevron-down"
                role="button"
              />
            </div>
          </div>
        </div>
      </div>
      <section
        className="m-paragraph project-desc"
      >
        <p
          className="m-paragraph-line "
        >
          This is MLReefÂ´s demo repository.
        </p>
      </section>
      <div
        className="feature-list"
      >
        <a
          className="feature"
          href="/my-namespace/the-project-name"
          id="data"
          onClick={[Function]}
        >
          Data
        </a>
        <a
          className="feature"
          href="/my-projects/12395599/insights/-/jobs"
          id="insights"
          onClick={[Function]}
        >
          Insights
        </a>
      </div>
    </div>
  </div>
  <div
    className="main-content"
  >
    <div
      className="epmty-experiment-logo"
    >
      <img
        alt="Create an experiment"
        src={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        width="240"
      />
      <span>
        You don't have any experiment in your ML project
      </span>
      <p>
        Why not start one?
      </p>
      <button
        className=" btn btn-primary"
        disabled={false}
        id="new-experiment"
        onClick={[Function]}
        type="button"
        variant="contained"
      >
        Start an experiment
      </button>
    </div>
  </div>
  <br />
  <br />
</div>
`;
